src/services/catalog/app.py

```diff
@@ -73,8 +74,30 @@ def GET(self, *path, **args):
 
         elif path[0] == "services":
 
+            if "type" in args:
+                # /catalog/services?type={service,device}&subtype={arduino,raspberry}
+                try:
+                    stype = ServiceType.value_of(str(args["type"]).upper())
+                except ValueError:
+                    cherrypy.response.status = 400
+                    return self.asjson_error("Invalid service type")
+
+                sers = [s for s in self._serviceManager.services.values() if s.stype == stype]
+                if "subtype" in args:
+                    try:
+                        subtype = ServiceSubType.value_of(str(args["subtype"]).upper())
+                    except ValueError:
+                        cherrypy.response.status = 400
+                        return self.asjson_error("Invalid service subtype")
+
+                    return self.asjson({"services": [s.toDict() for s in sers if s.subtype == subtype]})
+
+                # /catalog/services?type={service,device}
+                else:
+                    return self.asjson({"services": [s.toDict() for s in sers]})
+
```

Comment:
As I alredy asked here, https://github.com/thegabriele97/WaterIoT/pull/10#discussion_r819767674, I don't get these lines of code. The answer to my doubt can be found here: https://github.com/thegabriele97/WaterIoT/pull/10#discussion_r819767674

Reply:
Answered in https://github.com/thegabriele97/WaterIoT/pull/10#discussion_r819813800